### PR TITLE
Tweak Ubuntu source install

### PR DESCRIPTION
**Pull request by chapulina**
I was setting up a new machine and noticed some places that could use improvements.

* made Citadel, Dome and Edifice installation consistent
* removed Ignition and SDF libraries from the ones installed from `packages.apt` (fixes https://github.com/ignitionrobotics/docs/issues/119)
* Added some generic tools up top which are needed for the tutorial to work on a fresh install

### CODE REVIEW

File: edifice/install_ubuntu_src.md [hunk 1]
```diff
@@ -217,35 +224,4 @@ the results you want:
...
-To enable Ogre 2 support, you'll need to update your computer's OpenGL version.
-As suggested on the Ogre logs, this may require updating your graphics card
-drivers.
-
-The Ogre 2 debs from the osrfoundation repository are built from a fork of
-Ogre's `v2-1` branch with changes needed for deb packaging and allowing it to
-be co-installable with Ogre 1.x. The code can be found here:
-
-https://github.com/osrf/ogre-2.1-release
+See [Troubleshooting](troubleshooting)
```
Comment (ahcorde) [inline edifice/install_ubuntu_src.md:227]: Did you forget to add this file ? I cannot see this file in the edifice folder.

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: edifice/install_ubuntu_src.md (hunk 1 above)
Comment (chapulina) [reply edifice/install_ubuntu_src.md:227]: Nice catch, added: 7c8a3ee36dfa875403c2d096be73bc356747b5e7

File: edifice/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    glxinfo | grep "OpenGL version"
+
+You should be able to use Ogre 1 without any issues however. You can check if
+that's working by running a world which uses Ogre 1 instead of Ogre 2, such as:
+
+    ign gazebo -v 3 lights.sdf
+
+If that loads, you can continue to use Ignition with Ogre 1, just be sure to
+specify `ogre` in your SDF files instead of `ogre2`.
```
Comment (ahcorde) [inline edifice/troubleshooting.md:81]: what about including here the option too ? `--render-engine ogre`

**Review by ahcorde — APPROVED** [review]
small comment in the troubleshooting section

File: edifice/troubleshooting.md (hunk 2 above)
Comment (chapulina) [reply edifice/troubleshooting.md]: Totally! That's way easier than changing the SDF file: b61127ebca76d339c1a6fec464819d4e63df8a8c

**Comment by mabelzhang** [comment]
I ran the latest tutorial in a Docker container, and the dependencies line seems not very reliable.
```
$ sudo apt -y install \
>   $(sort -u $(find . -iname 'packages-'`lsb_release -cs`'.apt' -o -iname 'packages.apt') | sed '/ignition\|sdf/d' | tr '\n' ' ')
E: Command line option '0' [from -0400] is not understood in combination with the other options.
```

When I run just the `sort` part, I get:
```
$ sort -u $(find . -iname 'packages-'`lsb_release -cs`'.apt' -o -iname 'packages.apt') | sed '/ignition\|sdf/d' | tr '\n' ' '
0000000000000000000000000000000000000000 7a07a71f6b369aa2fa6223abab80f61001417cb7 Mabel Zhang <mabel@openrobotics.org> 1603134832 -0400	pull: storing head 7a07a71f6b369aa2fa6223abab80f61001417cb7 binutils-dev build-essential cmake curl freeglut3-dev libavcodec-dev libavdevice-dev libavformat-dev libavutil-dev libbenchmark-dev libcurl4-openssl-dev libdart-collision-ode-dev libdart-dev libdart-external-ikfast-dev libdart-external-odelcpsolver-dev libdart-utils-urdf-dev libeigen3-dev libfreeimage-dev libgflags-dev libglew-dev libgts-dev libjsoncpp-dev libogre-1.9-dev libogre-2.1-dev libprotobuf-dev libprotoc-dev libsqlite3-dev libswscale-dev libtinyxml2-dev liburdfdom-dev libwebsockets-dev libxi-dev libxml2-utils libxmu-dev libyaml-dev libzip-dev libzmq3-dev pkg-config protobuf-compiler python-psutil python-yaml qml-module-qt-labs-folderlistmodel qml-module-qt-labs-platform qml-module-qt-labs-settings qml-module-qtcharts qml-module-qtgraphicaleffects qml-module-qtqml-models2 qml-module-qtquick-controls qml-module-qtquick-controls2 qml-module-qtquick-dialogs qml-module-qtquick-layouts qml-module-qtquick-templates2 qml-module-qtquick-window2 qml-module-qtquick2 qtbase5-dev qtdeclarative5-dev qtquickcontrols2-5-dev rubocop ruby ruby-dev ruby-ronn swig uuid-dev xvfb
```

I'm not sure why it has all the junk in the beginning. The effect is that the `sudo apt` line doesn't install anything, and I get compiler errors about ruby, eigen3, protobuf, etc., everything missing.

**Comment by chapulina** [comment]
Ohh maybe it needs this fix?

https://github.com/ignition-tooling/release-tools/pull/323

**Comment by mabelzhang** [comment]
Ah, that does look a lot better:
```
$ sort -u $(find . -iname 'packages-'`lsb_release -cs`'.apt' -o -iname 'packages.apt' | grep -v '/\.git/') | sed '/ignition\|sdf/d' | tr '\n' ' '
binutils-dev curl freeglut3-dev libavcodec-dev libavdevice-dev libavformat-dev libavutil-dev libbenchmark-dev libcurl4-openssl-dev libdart-collision-ode-dev libdart-dev libdart-external-ikfast-dev libdart-external-odelcpsolver-dev libdart-utils-urdf-dev libeigen3-dev libfreeimage-dev libgflags-dev libglew-dev libgts-dev libjsoncpp-dev libogre-1.9-dev libogre-2.1-dev libprotobuf-dev libprotoc-dev libsqlite3-dev libswscale-dev libtinyxml2-dev liburdfdom-dev libwebsockets-dev libxi-dev libxml2-utils libxmu-dev libyaml-dev libzip-dev libzmq3-dev pkg-config protobuf-compiler python-psutil python-yaml qml-module-qt-labs-folderlistmodel qml-module-qt-labs-platform qml-module-qt-labs-settings qml-module-qtcharts qml-module-qtgraphicaleffects qml-module-qtqml-models2 qml-module-qtquick-controls qml-module-qtquick-controls2 qml-module-qtquick-dialogs qml-module-qtquick-layouts qml-module-qtquick-templates2 qml-module-qtquick-window2 qml-module-qtquick2 qtbase5-dev qtdeclarative5-dev qtquickcontrols2-5-dev ruby ruby-dev ruby-ronn swig uuid-dev xvfb
```

**Comment by mabelzhang** [comment]
#141

**Comment by mabelzhang** [comment]
Ah I forgot to say, there was still some packages not installed after all the above.
`python-psutil` was not enough, I think, because I still got:
```
--- stderr: sdformat10                                                                                                                                                                                     
CMake Warning at CMakeLists.txt:214 (message):
  -- BUILD WARNINGS
CMake Warning at CMakeLists.txt:216 (message):
  -- 	Python psutil package not found.  Memory leak tests will be skipped
CMake Warning at CMakeLists.txt:218 (message):
  -- END BUILD WARNINGS
```
It went away after I installed `python3-psutil`. This was Ubuntu 20.04.

Similarly for `doxygen`. I got both an error and a warning from `sdformat/doc/CMakeLists.txt`:
```
CMake Warning at /usr/share/cmake-3.16/Modules/FindDoxygen.cmake:447 (message):
  Unable to determine doxygen version: No such file or directory
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindDoxygen.cmake:610 (_Doxygen_find_doxygen)
  doc/CMakeLists.txt:1 (find_package)

CMake Error at /usr/share/cmake-3.16/Modules/FindDoxygen.cmake:689 (message):
  Unable to generate Doxyfile template: No such file or directory
Call Stack (most recent call first):
  doc/CMakeLists.txt:1 (find_package)
```

It looks like the command does look for sdf... so I'm not sure how to get this into the list programmatically.

**Comment by chapulina** [comment]
> there was still some packages not installed after all the above.

I believe the idea is to leave the optional tooling packages out, because most users will not want them. If someone wants to run memory leak tests or generate docs, they should install those dependencies separately.

**Comment by mabelzhang** [comment]
Then the Doxygen should be a warning, not an error, right?

The `sort` command also outputted `python-psutil`, which is only needed for sdformat's cmake/SearchForStuff.cmake . That one is a warning, but we are already `sudo apt install`ing `python-psutil`, we should also (or instead) install `python3-psutil`. Otherwise that package we get the user to install is not doing anything right now.

**Comment by chapulina** [comment]
> Then the Doxygen should be a warning, not an error, right?

Oh yeah I didn't realize it was an error. It's interesting that CI is not complaining, I wonder what's the difference in the setup. Doxygen version perhaps?

> we are already sudo apt installing python-psutil, we should also (or instead) install python3-psutil

You're right, I think `python3-psutil` may work from Bionic, so it could just be updated here:

https://github.com/osrf/sdformat/blob/master/.github/ci/packages.apt

**Pull request merged**